### DISCUSSION TITLE
Counters and cluster-wide transaction

### DISCUSSION THREAD

**Discussion by georgiosd** [ravendb/ravendb#15186] [Q&A] [answered]
Hi!

We just came across this exception for the first time, and I'd like to better understand it:

```
System.NotSupportedException: The document <id> has counters, this is not supported in cluster wide transaction.
```

First of all, why is it thrown when we haven't modified the counters at all?

Secondly, what is the recommended practice in this case? Don't attach counters to any document that has to be modified with a cluster-wide transaction?

**Comment by garayx** [comment]
Hi,

Yes, adding documents with counter / attachment / times series in cluster wide transaction is not supported.

Can you share a repro for this exception when you don't add counters ?

**Comment by ayende** [comment, marked as answer]
The reasoning is that it messes up some (edge cases).

Counters cannot be modified in a cluster wide transaction, but the _metadata_ for counters is important.
If we didn't block that, then you may end up in a situation where in a conflict between a cluster wide transaction and a single node transaction (that added a counter).
This results in a conflict, that resolves in favor of the cluster wide version. But the metadata on the document will not include the counter, so it is now "hidden".

The way to manage that is to have `users/1-A` that you modify via cluster wide transaction and split timeseries / counters etc to `users/1-A/counters`
